packages/chain-bridge/src/adapters/LayerZeroAdapter.sol

```diff
@@ -87,6 +101,35 @@ contract LayerZeroAdapter is
         }
     }
 
+    /*//////////////////////////////////////////////////////////////
+                             GOVERNANCE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the OFT contract address for a token
+     * @param token The ERC20 token address
+     * @param oft The LayerZero OFT contract address for this token
+     */
+    function setOftForToken(address token, address oft) external onlyGovernor {
+        if (token == address(0) || oft == address(0)) revert InvalidParams();
+
+        // Validate OFT configuration - supports both adapter and direct patterns
+        // Adapter pattern: IOFT(oft).token() returns the underlying ERC20
+        // Direct pattern: IOFT(oft).token() returns oft itself, or may not exist
+        try IOFT(oft).token() returns (address t) {
+            // For adapter pattern: token must match the underlying asset
+            // For direct pattern: token must equal oft
+            if (t != token && t != oft) revert InvalidParams();
+        } catch {
+            // Direct OFT may not have token() function
```

Comment:
If the token inherits OFT or implements IOFT, it will have the token() function. We might even be able to skip the try...catch and just let it revert, or explicitly revert within the catch block.